src/proc.cpp

```diff
@@ -754,19 +753,17 @@ static void read_try(job_t *j) {
     }
 }
 
-/// Give ownership of the terminal to the specified job.
-///
-/// \param j The job to give the terminal to.
-/// \param cont If this variable is set, we are giving back control to a job that has previously
-/// been stopped. In that case, we need to set the terminal attributes to those saved in the job.
-bool terminal_give_to_job(const job_t *j, bool cont) {
-    errno = 0;
+// Return control of the terminal to a job's process group. restore_attrs is true if we are restoring
+// a previously-stopped job, in which case we need to restore terminal attributes.
+bool terminal_give_to_job(const job_t *j, bool restore_attrs) {
     if (j->pgid == 0) {
         debug(2, "terminal_give_to_job() returning early due to no process group");
         return true;
     }
 
-    signal_block();
+    // RAII wrappers must have a name so that their scope is tied to the function as it is legal for
+    // the compiler to construct and then immediately deconstruct unnamed objects otherwise.
+    signal_block_t signal_block;
```

Comment:
This stuff is nice!

But let me just say that I think whoever named a bunch of C++ idioms has no naming skills whatsoever. RAII is, as a term, crap.

Reply:
😄 what would you do if I threw in a SFINAE or two in there?